src/TorchSharp/DataLoader.cs

```diff
@@ -412,35 +411,42 @@ public bool MoveNext()
                 {
                     DisposeCurrent();
 
-                    using (var scope = torch.NewDisposeScope()) {
```

Comment:
I just realized that the dispose scope does not work on other threads. Instead dispose scopes should be separately created in the below `ForAll`.